packages/grid/x-grid-modules/src/hooks/features/useColumnReorder.tsx

```diff
@@ -0,0 +1,65 @@
+import * as React from 'react';
+import { ColDef } from '../../models/colDef';
+import { useLogger } from '../utils';
+import { ApiRef } from '../../models';
+
+export const useColumnReorder = (
+  apiRef: ApiRef
+) => {
+  const logger = useLogger('useColumnReorder');
+
+  const dragCol = React.useRef<ColDef | null>();
+  const dragColNode = React.useRef<HTMLElement | null>();
+
+  const handleDragStart = React.useCallback(
+    (col: ColDef, htmlEl: any): void => {
+      logger.debug(`Start dragging col ${col.field}`);
+      dragCol.current = col;
+      dragColNode.current = htmlEl;
+      dragColNode.current?.addEventListener('dragend', handleDragEnd);
+      dragColNode.current?.classList.add('dragging');
+      setTimeout(() => {
+        dragColNode.current?.classList.remove('dragging');
```

Comment:
Why not removing the class in the `dragend`?

Reply:
This. is a needed hack unfortunately - the problem is with how HTML5 drag API works. When you start dragging it selects the current DOM element with all its styles, but what we want is for the thing we are actually dragging (the thing that follows the mouse cursor) to have a style different than its original DOM element. That's why we need to style it for one tick, so that the dragging starts and then remove the style from the original element. That way the DOM will be repainted but the element we are dragging will have the new styles.

Hope I managed to explain it.

If we remove the style in the `dragend` then during the dragging both the original DOM element and the one we are dragging will have the same styles